### PR TITLE
Geometry_Engine: issue 460, Common_Engine: issue 902: Join (exponential growth of time), Normal (fails on circular panels)

### LINKED ISSUE #460: Geometry_Engine: Join - Exponential growth of computing time
### This issue has been archived 

Historic issue archive can be found [here](https://burohappold.sharepoint.com/sites/BHoM/01_Master%20File/Forms/AllItems.aspx?id=%2Fsites%2FBHoM%2F01_Master%20File%2F999_BHoM_GitHub_History_Archive)

### LINKED ISSUE #902: Common_Engine : Query.Normal fails on Panels with Circular Boundaries
See failing test file [here](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?RootFolder=%2Fsites%2FBHoM%2F02_Current%2F12_Scripts%2F01_Test%20Scripts%2FBHoM_Engine%2FCommon_Engine%2FCommon_Engine-Issue902-NormalQueryOnCircularPanel&FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4)

### PR DESCRIPTION
**Pull request by LMarkowski**
### Issues addressed by this PR
Closes #460 
Closes #902 

In #460 I did exactly what @epignatelli suggested in archived comments [here](https://burohappold.sharepoint.com/sites/BHoM/01_Master%20File/Forms/AllItems.aspx?id=%2Fsites%2FBHoM%2F01_Master%20File%2F999_BHoM_GitHub_History_Archive). Now Join runs ok. Former method was adding more and more points to the list while recalculating.

In #902 it was caused by geometry engine method which did not cover polycurves consisting of 1 curve being circle or ellipse. Now it does.

### Test files
#460:
[Here](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4&id=%2Fsites%2FBHoM%2F02_Current%2F12_Scripts%2F01_Test%20Scripts%2FBHoM_Engine%2FGeometry_Engine%2FGeometry_Engine-issue460-Join%20-ExponentialGrowthOfComputingTime) is issue specific test. I also updated normal test in [here](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4&id=%2Fsites%2FBHoM%2F02_Current%2F12_Scripts%2F01_Test%20Scripts%2FBHoM_Engine%2FGeometry_Engine%2FModify)

#902:
[Here](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?RootFolder=%2Fsites%2FBHoM%2F02_Current%2F12_Scripts%2F01_Test%20Scripts%2FBHoM_Engine%2FCommon_Engine%2FCommon_Engine-Issue902-NormalQueryOnCircularPanel&FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4) is issue specific test which only proves that Common Engine method returns vectors. Proof of geometrical correctness is in normal tests [here](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4&id=%2Fsites%2FBHoM%2F02_Current%2F12_Scripts%2F01_Test%20Scripts%2FBHoM_Engine%2FGeometry_Engine%2FQuery)

### Changelog
Updated:

- public static List<Polyline> Join(List<Polyline>, tolerance)
-  public static Vector Normal(PolyCurve)

### CODE REVIEW

**Comment by FraserGreenroyd** [comment]
Discussion on BHoM call: deprecate `modify` `join` and move it to `Compute` instead as part of this PR :smile:

**Comment by pawelbaran** [comment]
I've added to the original test file by @alexradne to the [test folder](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4&id=%2Fsites%2FBHoM%2F02_Current%2F12_Scripts%2F01_Test%20Scripts%2FBHoM_Engine%2FGeometry_Engine%2FGeometry_Engine-issue460-Join%20-ExponentialGrowthOfComputingTime).

**Comment by LMarkowski** [comment]
Moved join to Compute, added Deprecated Attributes in Modify and added test file in Compute

**Comment by LMarkowski** [comment]
@IsakNaslundBh To move Join to Compute I had to make some cosmetic changes in methods in Structure. Is it ok or Structure is still blocked?

File: Geometry_Engine/Modify/Join.cs [hunk 1]
```diff
@@ -33,158 +34,165 @@ public static partial class Modify
         /****                Join curves                ****/
         /***************************************************/
 
+        [DeprecatedAttribute("2.3", "Moved to Compute", typeof(Compute), "IJoin")]
         public static List<PolyCurve> IJoin(this List<ICurve> curves, double tolerance = Tolerance.Distance)
         {
-            List<PolyCurve> sections = curves.Select(c => new PolyCurve { Curves = new List<ICurve> { c.IClone() } }).ToList();
+            //List<PolyCurve> sections = curves.Select(c => new PolyCurve { Curves = new List<ICurve> { c.IClone() } }).ToList();
```
Comment (IsakNaslundBh) [inline Geometry_Engine/Modify/Join.cs]: Could we please just delete the commented out code here?

**Comment by pawelbaran** [comment]
Parked until @LMarkowski checks dependencies

**Comment by LMarkowski** [comment]
Found Dependencies and fixed them in PR: [#305](https://github.com/BHoM/Revit_Toolkit/pull/305) and [#145](https://github.com/BuroHappoldEngineering/ModelLaundry_Toolkit/pull/145)

**Pull request merged**